src/main/java/appeng/api/stacks/AEKey.java

```diff
@@ -242,7 +263,10 @@ public final boolean supportsFuzzyRangeSearch() {
         return getType().supportsFuzzyRangeSearch();
     }
 
-    public abstract Component getDisplayName();
+    public Component getDisplayName() {
+        assert this.displayName != null;
```

Comment:
Please use `Objects.requireNonNull(this.displayName)`, java assertions are useless since they are never enabled in practice.